projects/transformers/callbacks/sparsity.py

```diff
@@ -49,3 +59,40 @@ def on_init_end(self, args, state, control, model, **kwargs):
     def on_step_end(self, args, state, control, model, **kwargs):
         """Rezero weights"""
         model.apply(rezero_weights)
+
+
+class RezeroWeightsCallback(TrainerCallback):
+
+    def on_init_end(self, args, state, control, model, **kwargs):
+        """Log sparsity of the model and the sparsity of just the encoder."""
+
+        model.apply(rezero_weights)
+
+        num_tol, num_nonzero = count_nonzero_params(model)
```

Comment:
Is num_tol a typo? It feels like you wanted to write num_total

Reply:
Tried to go for a shorthand but messed up there.